ddp-workspace/projects/ddp-dsm-ui/src/app/sharedLearningUpload/components/filesTable/filesTable.component.spec.ts

```diff
@@ -0,0 +1,234 @@
+  });
+
+  it('should display send to participant SUCCESS',async () => {
+    const sendToParticipantSuccessDocument: SomaticResultsFileWithStatus = cloneDeep(testDocuments[0]);
+    sendToParticipantSuccessDocument.sendToParticipantStatus.status = HttpRequestStatusEnum.SUCCESS;
+    component.somaticResultsFiles = [sendToParticipantSuccessDocument];
+    fixture.detectChanges();
+    const status = await getDocumentCellData(0, ColumnNameEnum.SEND_TO_PARTICIPANT);
+    expect(status).toEqual(DocumentIconsEnum.SUCCESS, 'Wrong send to participant status');
+  });
+
+  it('should display send to participant FAIL',async () => {
+    const sendToParticipantFailDocument: SomaticResultsFileWithStatus = cloneDeep(testDocuments[0]);
+    sendToParticipantFailDocument.sendToParticipantStatus.status = HttpRequestStatusEnum.FAIL;
+    component.somaticResultsFiles = [sendToParticipantFailDocument];
+    fixture.detectChanges();
+    const status = await getDocumentCellData(0, ColumnNameEnum.SEND_TO_PARTICIPANT);
+    expect(status).toEqual(DocumentIconsEnum.FAIL, 'Wrong send to participant status');
+  });
+
+  it('should display send to participant IN PROGRESS',async () => {
+    const sendToParticipantInProgressDocument: SomaticResultsFileWithStatus = cloneDeep(testDocuments[0]);
+    sendToParticipantInProgressDocument.sendToParticipantStatus.status = HttpRequestStatusEnum.IN_PROGRESS;
+    component.somaticResultsFiles = [sendToParticipantInProgressDocument];
+    fixture.detectChanges();
+    const status = await getDocumentCellData(0, ColumnNameEnum.SEND_TO_PARTICIPANT);
+    expect(status).toEqual(DocumentIconsEnum.SPINNING, 'Wrong send to participant status');
+  });
+
+  it('should display delete IN PROGRESS',async () => {
+    const deleteInProgressDocument: SomaticResultsFileWithStatus = cloneDeep(testDocuments[0]);
+    deleteInProgressDocument.deleteStatus.status = HttpRequestStatusEnum.IN_PROGRESS;
+    component.somaticResultsFiles = [deleteInProgressDocument];
+    fixture.detectChanges();
+    const status = await getDocumentCellData(0, ColumnNameEnum.DELETE);
+    expect(status).toEqual(DocumentIconsEnum.SPINNING, 'Wrong delete status');
+  });
+
+
+  /* HELPER FUNCTIONS */
+
+  const getDocumentCellData = async (rowIndex: number, columName: ColumnNameEnum): Promise<DocumentIconsEnum> => {
+    const matTableHarness = await materialHarnessLoader.getMatTableHarness();
+    const rows = await matTableHarness.getRows();
+    // the status information cell
+    const cells = await rows[rowIndex].getCells();
+    // cell data
+    let cellData = '';
+
+    // looking for cell data for the provided column
+    for(const cell of cells) {
+      if(await cell.getColumnName() === columName) {
+        cellData = await cell.getText();
+        break;
+      }
+    }
+
+    // returns either matIcon text or SCANNING (matSpinner doesn't have text)
+    return (cellData  || 'SPINNING') as DocumentIconsEnum;
```

Comment:
How about this new assertion, `expect(cellData).not.toEqual('')` in case `cells.length === 0`?